src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodDriver : GpioDriver
+    {
+        private const string GpioGet = "gpioget";
+        private const string GpioChip0 = "gpiochip0 ";
+        private const string GpioSet = "gpioset";
+        public int TimeoutMilliSeconds { get; set; } = (int)TimeSpan.FromSeconds(1).TotalMilliseconds;
+        private class MutableTuple<Item1, Item2>
+        {
+            public Item1 Value1 { get; set; }
+            public Item2 Value2 { get; set; }
+            public MutableTuple(Item1 item1, Item2 item2) {
+                Value1 = item1;
+                Value2 = item2;
+            }
+        }
+        private Dictionary<int, MutableTuple<PinMode, Process>> _pinNumberToPinModeAndProcess = new Dictionary<int, MutableTuple<PinMode, Process>>();
+        protected internal override int PinCount => throw new NotImplementedException();
+
+        protected internal override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventType, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void ClosePin(int pinNumber)
+        {
+            MutableTuple<PinMode, Process> pinModeAndProcess;
+            if (_pinNumberToPinModeAndProcess.TryGetValue(pinNumber, out pinModeAndProcess)) 
+            {
+                Process existingProcess = pinModeAndProcess.Value2;
+                if (existingProcess != null) {
+                    existingProcess.Kill();
+                    existingProcess.WaitForExit(1);
```

Comment:
Why 1 ms specifically? Seems not worth the bother. I suggest to wait forever, or wait for some reasonable timeout (a few seconds?) that is stored in a const int somewhere.

Reply:
Maybe TimeoutMilliSeconds

Reply:
I assumed that killing a process wouldn't take a long so used 1 milliseconds, was thinking about waiting forever but was not sure if the WaitForExit() method would exit after process killed or wait until the time set is elapsed, didn't found exact answer from google … some was sayin it will still wait even the process is done

Reply:
On a heavily loaded, or overwhelmed machine, it could take an arbitrary time. Minutes, even. So you should not consider "how long it normally takes" but "how long am I prepared to wait" and "is it a failure if it doesn't happen". 

If it's a failure if it doesn't happen, I would make the wait quite large - maybe 10-30 seconds, and possibly infinite.

I'm guessing since this is IoT stuff these won't ever run in CI, but note that our CI and official build runs can sometimes run on heavily loaded/slow machines (lots of tests in parallel) so in tests we use timeouts far longer than one might think. For example for something that usually takes a second on a dev machine, a typical timeout we use in a test might be 30 seconds. Less than that and it fails every week or two.

Reply:
As we cannot set a value to a pin until the process associated to that pin completed waiting indefinitely might be better I will test with that

Reply:
Test result for using indefinite wait (without setting any milliseconds) was similar as waiting 1 milliseconds, so keeping it with indefinite wait